portal/src/main/webapp/js/src/co-exp/coExpression.js

```diff
@@ -275,52 +334,179 @@ var CoExpView = (function() {
+                                    case_ids_key: window.QuerySession.getCaseIdsKey(),
+                            };
+                            $.post("getCoExp.do", paramsGetCoExpData, function(result) {
+                                convertData(result, geneEntityId, "GENESET");
+                                if (genesetMap[cbio.util.safeProperty(geneEntityId)].length >= 1) { //Only add gene sets if we have them
+                                    $("#" + Names.tableId).dataTable().fnAddData(genesetMap[cbio.util.safeProperty(geneEntityId)]);
+                                    } else {
+                                        document.getElementById("no_genesets"+Names.tableDivId).style.display = 'block'
+                                    }
+                            }, "json");
+                        }
+                    } else { //Checkbox not checked, clear the whole table
+                        $("#" + Names.tableId).dataTable().fnClearTable();
+                        document.getElementById("no_genesets"+Names.tableDivId).style.display = 'none'
+                        if ($("#gene_checkbox"+cbio.util.safeProperty(geneEntityId)).prop('checked')) { //If the gene box is checked, keep the genes
+                            if (geneMap.hasOwnProperty(cbio.util.safeProperty(geneEntityId)) && geneMap.hasOwnProperty(cbio.util.safeProperty(geneEntityId)) && geneMap[cbio.util.safeProperty(geneEntityId)].length >= 1) {
+                                $("#" + Names.tableId).dataTable().fnAddData(geneMap[cbio.util.safeProperty(geneEntityId)]);
+                            }
+                        }
+                   }
+               });
+           }
 
             function attachRowListener() {
                 $("#" + Names.tableId + " tbody tr").live('click', function (event) {
                     //Highlight selected row
-                    $(coExpTableInstance.fnSettings().aoData).each(function (){
+                    $($("#" + Names.tableId).dataTable().fnSettings().aoData).each(function (){
                         $(this.nTr).removeClass('row_selected');
                     });
                     $(event.target.parentNode).addClass('row_selected');
                     //Get the gene name of the selected row
-                    var aData = coExpTableInstance.fnGetData(this);
+                    var aData = $("#" + Names.tableId).dataTable().fnGetData(this);
                     if (null !== aData) {
                         $("#" + Names.plotId).empty();
                         $("#" + Names.plotId).append("<img style='padding:220px;' src='images/ajax-loader.gif' alt='loading' />");
                         var coexpPlots = new CoexpPlots();
-                        coexpPlots.init(Names.plotId, geneId, aData[0], aData[2], aData[3], $("#coexp-profile-selector :selected").val());
+                        var profile1Id = null;
```

Comment:
Since this leads to the building of the co-expression plots, I renamed the variables `entity_x_profile` and `entity_y_profile`, and in `GetGeneticAlterationDataJSON` I also renamed `entity1` and `entity2` to `entity_x` and `entity_y`. This names make more sense.